app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,47 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def edit
+    load_partner
+    load_associations
+  end
+
+  def update
+    load_partner
+    build_partner
+    save_partner or render 'edit'
+  end
+
+  private
+  def load_partner
+    @partner ||= Partner.find(params[:id])
+  end
```

Comment:
this method returns a partner or throws an errors. We should return partner or nil.